File: android/data/LocalBroadcast_2017_6_22/app/src/main/java/top/goluck/localbroadcast_2017_6_22/MainActivity.java
package top.goluck.localbroadcast_2017_6_22;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.support.v4.content.LocalBroadcastManager;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView text;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        text = (TextView) findViewById(R.id.text);
    }

    @Override
    protected void onResume() {
        super.onResume();
        // 注册本地广播
        IntentFilter filter = new IntentFilter(WorkService.WORK_ACTION);
        LocalBroadcastManager.getInstance(this).registerReceiver(mReceiver, filter);
    }

    @Override
    protected void onPause() {
        super.onPause();
        // 解绑本地广播
        LocalBroadcastManager.getInstance(this).unregisterReceiver(mReceiver);
    }

    public void onAction(View view){
        switch (view.getId()){
            case R.id.button:
                Log.i("msg","启动了服务");
                // 启动服务发送广播
                Intent intent = new Intent(this, WorkService.class);
                intent.putExtra("DATA",text.getText().toString().length()>800?"Hello World":text.getText().toString());
                startService(intent);

                sendBroadcast(new Intent("top.goluck.receiver.action"));
                break;
        }
    }

    /**
     * 广播接收器
     */
    private BroadcastReceiver mReceiver = new BroadcastReceiver() {

        @Override
        public void onReceive(Context context, Intent intent) {
            Log.i("msg","接收到了广播");
            String date = intent.getStringExtra("DATA");
            String time = intent.getStringExtra("TIME");
            text.setText(date + "\n" + time);
        }
    };

}

File: android/ui/AppLinks_2018_10_7/app/src/main/java/top/goluck/applinks_2018_10_7/MainActivity.kt
package top.goluck.applinks_2018_10_7

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        webview.loadUrl("file:///android_asset/index.html")
        getTopIntent()
    }

    fun getTopIntent(){
        if (null != intent && null != intent.data) {
            // uri 就相当于 web 页面中的链接
            var uri = intent.data
            text.text = "uri = " +uri
            text.text = text.text.toString() + "\n scheme = " + uri.scheme
            text.text = text.text.toString() + "\n host = " + uri.host
            text.text = text.text.toString() + "\n key1 = " + uri.getQueryParameter("key1")
            text.text = text.text.toString() + "\n key2 = " + uri.getQueryParameter("key2")
        }
    }

}